src/linter/block.go

```diff
@@ -1471,6 +1475,18 @@ type andWalker struct {
 
 func (a *andWalker) EnterNode(w walker.Walkable) (res bool) {
 	switch n := w.(type) {
+	case *expr.FunctionCall:
+		args := n.ArgumentList.Arguments
+		nm, ok := n.Function.(*name.Name)
+		if ok && meta.NameToString(nm) == `method_exists` && len(args) == 2 {
```

Comment:
I would suggest to use NameEquals to avoid unnecessary allocations in hot paths

Reply:
Maybe we can re-write NameEquals to be more efficient?

I think it also does allocations (it does strings.Split which allocates a slice).

Reply:
Right now it doesn't seem to be faster:
```
BenchmarkNameEquals-8  13470901 79.3 ns/op 16 B/op  1 allocs/op
BenchmarkToString-8    19985751 59.1 ns/op 16 B/op  1 allocs/op
```
```go
var theName = &name.Name{
	Parts: []node.Node{
		&name.NamePart{Value: `method_exists`},
	},
}

func BenchmarkNameEquals(b *testing.B) {
	for i := 0; i < b.N; i++ {
		_ = NameEquals(theName, `method_exists`)
	}
}

func BenchmarkToString(b *testing.B) {
	for i := 0; i < b.N; i++ {
		_ = NameToString(theName) == `method_exists`
	}
}
```

Reply:
You're probably right, we need to optimise NameEquals if it does allocate memory. The reason why I still think that we should use NameEquals is that once we do optimise it, all the code that uses it will become faster. If we always use NameToString we can only optimise the use-case where there is only part. If there are two parts, for example, memory allocation will still be required.